src/blueprint/from_config/operators/call.rs

```diff
@@ -0,0 +1,79 @@
+use crate::blueprint::*;
+use crate::config;
+use crate::config::{Config, Field, GraphQLOperationType};
+use crate::directive::DirectiveCodec;
+use crate::try_fold::TryFold;
+use crate::valid::Valid;
+
+pub fn update_call(
+  operation_type: &GraphQLOperationType,
+) -> TryFold<'_, (&Config, &Field, &config::Type, &str), FieldDefinition, String> {
+  TryFold::<(&Config, &Field, &config::Type, &str), FieldDefinition, String>::new(
+    move |(config, field, type_of, name), b_field| {
+      let Some(call) = &field.call else {
+        return Valid::succeed(b_field);
+      };
+
+      if validate_field_has_resolver(name, field, &config.types).is_succeed() {
+        return Valid::fail(format!(
+          "@call directive is not allowed on field {} because it already has a resolver",
+          name
+        ));
+      }
+
+      let type_and_field = if let Some(query) = &call.query {
+        Valid::succeed(("Query", query.as_str()))
+      } else {
+        Valid::fail("call must have query".to_string())
+      };
+
+      type_and_field
+        .and_then(|(type_name, field_name)| {
+          Valid::from_option(
+            config.find_type(type_name),
+            format!("{} type not found on config", type_name),
+          )
+          .zip(Valid::succeed(field_name))
+        })
+        .and_then(|(query_type, field_name)| {
+          Valid::from_option(
+            query_type.fields.get(field_name),
+            format!("{} field not found", field_name),
+          )
+          .and_then(|field| {
+            if field.has_resolver() {
+              Valid::succeed((field, field_name, call.args.iter()))
+            } else {
+              Valid::fail(format!("{} field has no resolver", field_name))
+            }
+          })
+        })
+        .and_then(|(field, field_name, args)| {
+          args.fold(Valid::succeed(field.clone()), |field, (key, value)| {
+            field.and_then(|field| {
+              let value = value.replace("{{", "").replace("}}", "");
+
+              if let Some(http) = field.clone().http.as_mut() {
```

Comment:
we need to think about more general solution not tied to specific resolvers. It could be Expression that represents any resolving operation

Reply:
you mean creating a new variation under `Expression` enum that we'll handle on the resolvers? checking if I got it right.

Reply:
Is there any way to parse the `HashMap<String, String>` that currently represents `call.args` into a `BTreeMap<String, Arg>` (the type of `field.args`) so we can avoid creating a new expression?

Thinking here what would be the optimal solution to solve this issue. IMO this is one of the few things missing for this PR to be done.

CC: @tusharmath (in case you have any idea on how to solve it)